### PR TITLE
Add lesson on reconstruction basics

### PR DESCRIPTION
**Pull request by alexpearce**
The need for a new lesson near the beginning of the course was discussed during the [last meeting](https://indico.cern.ch/event/772256/). The consensus was that students should understand how the candidates they are filling ntuples with actually come into existence. This PR adds a lesson to try to help with that.

I've put the new lesson before the dataflow one. My idea was that we first discuss how candidates are built at a high level (this PR), then we dive in to how real data is processed, where reconstruction and candidate building plays a key role.

I'll soon update the dataflow lesson to lean more strongly on the concepts introduced here, e.g. explaining that the Stripping is just a bunch of these selections, and that's why we need to use decay descriptors (to select the decay(s) we care about from the big mixture).

I'd like to add a couple more figures, as this is otherwise a huge block of text. This would be a great place to add an explanation of the impact parameter, for example, and that's best done with a figure IMO. (In previous 'kits we've had to stop in the middle of the LoKI functors lesson to explain what IP is, what vertex fit quality is, etc.)

Any other suggestions of things to add here are very welcome, of course!

### CODE REVIEW

**Comment by vlisovsk** [comment]
Hmm, didn't you commit the same files in #141?

**Comment by alexpearce** [comment]
D'oh 🤦‍♂️ yes I did. I've removed the commit from #141, it should only be in this PR now.

**Comment by vlisovsk** [comment]
Thanks, really great job!
On the further ideas: probably one can add a sketch explaining what are "long/downstream/... tracks", and discuss in two sentences how the track is created from a set of hits.
So that the expression "track chi2" start making sense (I think someone asked me last year this question).

File: first-analysis-steps/physics-at-lhcb.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+a fit $$\chi^{2}$$ associated to it. The quality of the fit can be used in a
+selection.
+
+Finally, with the fitted muon momenta, we can form the four vector of the
+$$J/\psi$$ as their sum, creating the $$J/\psi$$ _candidate_[^1]. Now we get
+back to our original goal or measuring the number of true $$J/\psi \to
+\mu^{+}\mu^{-}$$. By plotting the $$J/\psi$$ invariant mass values as a
+histogram, we might hope to see a signal component.
+
+<a href="img/dimuon_mass.png"><img alt="Dimuon invariant mass spectrum" src="img/dimuon_mass.png" style="width: 50%; margin: 0 25%;"/></a>
```
Comment (vlisovsk) [inline first-analysis-steps/physics-at-lhcb.md]: Am I right that these mass plots are just toy data, not from any real LHCb dataset?

File: first-analysis-steps/physics-at-lhcb.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+We can then use model the components using probability density functions, and
```
Comment (vlisovsk) [inline first-analysis-steps/physics-at-lhcb.md]: This sentence looks not consistent. "We can then use model" -> "We can then model"?

File: first-analysis-steps/physics-at-lhcb.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+vertex. This is done by expressing the hypothesis that there is a common origin
+vertex of both tracks as an optimisation problem, and then varying the measured
+$$\mu^{+}$$ and $$\mu^{-}$$ four-momenta within their measured uncertainties to
+best fit that hypothesis. The result is a vertex object which has, for example,
+a fit $$\chi^{2}$$ associated to it. The quality of the fit can be used in a
+selection.
+
+Finally, with the fitted muon momenta, we can form the four vector of the
+$$J/\psi$$ as their sum, creating the $$J/\psi$$ _candidate_[^1]. Now we get
+back to our original goal or measuring the number of true $$J/\psi \to
```
Comment (vlisovsk) [inline first-analysis-steps/physics-at-lhcb.md]: or -> of?

File: first-analysis-steps/physics-at-lhcb.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+particles as ‘stable’. Unstable objects, with much shorter lifetimes, are
+formed as combinations of these ‘stable’ particles:
+
+1. Charged pions $$\pi^{\pm}$$
+2. Charged kaons $$K^{\pm}$$
+3. Protons $$p/\bar{p}$$
+4. Electrons $$e^{\pm}$$
+5. Muons $$\mu^{\pm}$$
+6. Photons $$\gamma$$
+
```
Comment (roelaaij) [inline first-analysis-steps/physics-at-lhcb.md]: A nit-picky person might include $$K_{L}$$ here, although we don't actually ever build these. I think they would look like a MIP in the calorimeter and then have muon hits.

Comment (vlisovsk) [reply first-analysis-steps/physics-at-lhcb.md]: By that logic, one should include deuterons too. Probably even neutrons. But it's better to keep things simple.

File: first-analysis-steps/physics-at-lhcb.md (hunk 1 above)
Comment (alexpearce) [reply first-analysis-steps/physics-at-lhcb.md]: That's right, I'll make a note of that.

File: first-analysis-steps/physics-at-lhcb.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+1. Select tracks created by the reconstruction;
+2. Create pairs of oppositely-charged tracks;
+3. Fit each pair under the hypothesis that they originate from a common point
+   in space.
+
+We’ll go over each of these steps, starting with tracks produced by the
+reconstruction. Naturally, there are many of these in any given event,
+typically hundreds, so we begin be applying _selections_ on the tracks based on
+our physics understanding. For example, the $$J/\psi$$ is quite heavy (at
+around $$3.1\,\mathrm{GeV}$$), so we might expect its decay products to have a
```
[the remark this answers is not in the document]
Comment (alexpearce) [reply first-analysis-steps/physics-at-lhcb.md]: Good catch, we should be consistent.

File: first-analysis-steps/physics-at-lhcb.md (hunk 4 above)
Comment (alexpearce) [reply first-analysis-steps/physics-at-lhcb.md]: I'd like to list everything that's in the output of the standard reconstruction. Deuterons have been present for a while so I'll add those. I'll add a footnote about neutrons and KLs (we should be nit-picky people :smile: ).

**Comment by vlisovsk** [comment]
Looks good to me! Thanks again for this amazing lesson!

**Comment by alexpearce** [comment]
No problem!

I'll add some more figures in a near-future PR.

**Comment by vlisovsk** [comment]
I am happy to merge it if no objections from anyone!

**Comment by vbellee** [comment]
No objection

**Pull request merged**